### PR TITLE
`fflas_ffpack`, `linbox`: Switch to release tarballs with working libtool

### PR DESCRIPTION
**Pull request by mkoeppe**
These fix the build on macOS, as discussed in #38020.

Test run: https://github.com/mkoeppe/sage/actions/runs/9142716213/job/25138580652

### :memo: Checklist

- [x] The title is concise and informative.
- [ ] The description explains in detail what this PR is about.
- [x] I have linked a relevant issue or discussion.
- [ ] I have created tests covering the changes.
- [ ] I have updated the documentation and checked the documentation preview.

### :hourglass: Dependencies

- Depends on #38020 (merged and partially reverted here)
- Depends on #37726 (merged here to resolve merge conflict)

### CODE REVIEW

**Comment by mkoeppe** [comment]
Tests (ci-37203-37834-37840-37919-37931-38008-38009-38020-38021-38025) at https://github.com/mkoeppe/sage/actions/runs/9142716213

File: build/pkgs/fflas_ffpack/checksums.ini [hunk 1]
```diff
@@ -1,5 +1,5 @@
...
-sha1=7c5faa81abc2b88ec24cec373b5e44cbaa7844dd
-md5=d8b7c113951a2a3f498a3aaadbe5620f
-cksum=3321469120
-upstream_url=https://github.com/linbox-team/fflas-ffpack/releases/download/vVERSION/fflas_ffpack-VERSION.tar.bz2
\ No newline at end of file
+tarball=fflas_ffpack-VERSION.tar.gz
+sha1=0beadd30b01dc5c890679aff73818a681513e82e
+md5=7b20fb95acb34d8d39fef2638cf2558e
+cksum=805176663
+upstream_url=https://github.com/mkoeppe/fflas-ffpack/releases/download/v2.5.0%2Bsage-2024-05-18b/fflas-ffpack-2.5.0.tar.gz
```
Comment (kwankyu) [inline build/pkgs/fflas_ffpack/checksums.ini:4]: Don't we have an "official" place under `https://github.com/sagemath/` to keep this tarball?

Can we keep the original url as commented out?

Comment (mkoeppe) [reply build/pkgs/fflas_ffpack/checksums.ini:4]: easier to just revert the commit when we can, I'd say

Comment (kwankyu) [reply build/pkgs/fflas_ffpack/checksums.ini:4]: We may need to keep these "in-house" tarballs for long time...

Comment (kwankyu) [reply build/pkgs/fflas_ffpack/checksums.ini:4]: Perhaps we fork `fflas-ffpack` under `sagemath` and ...

Comment (mkoeppe) [reply build/pkgs/fflas_ffpack/checksums.ini:4]: Well, I hope that my PRs that create releases will inspire upstream to make releases more often...

Comment (kwankyu) [reply build/pkgs/fflas_ffpack/checksums.ini:4]: > easier to just revert the commit when we can, I'd say

OK.

Comment (kwankyu) [reply build/pkgs/fflas_ffpack/checksums.ini:4]: > Well, I hope that my PRs that create releases will inspire upstream to make releases more often...

They do not seem to be active enough to be aware of your PRs. But let's see.

Your repo is OK for now.

**Comment by kwankyu** [comment]
This works well. Together with https://github.com/sagemath/sage/pull/38021, the build succeeded on my system.

**Review by kwankyu — APPROVED** [review]
LGTM.

**Comment by mkoeppe** [comment]
Thank you! Also one of the CI runs just completed: https://github.com/mkoeppe/sage/actions/runs/9142716213/job/25139831922

**Pull request merged**